src/Fantomas.Core/CodePrinter.fs

```diff
@@ -3458,7 +3458,10 @@ let genField (node: FieldNode) =
         | Some name ->
             genSingleTextNode name
             +> sepColon
-            +> autoIndentAndNlnIfExpressionExceedsPageWidth (genType node.Type))
+            +> ifElseCtx
```

Comment:
You want to introduce a new function similar to `autoIndentAndNlnExpressUnlessStroustrup` but which takes a `Type` instead of an `Expr`. Accepting any single `Type` will eventually lead to invalid code.

Example:
```fsharp
type MangaDexAtHomeResponse = {
    baseUrl: string
    chapter: {|
            hash: string
            data: string[]
            otherThing: int
        |}
        ->
            // comment
            (int -> int) -> {|
            hash: string
            data: string[]
            otherThing: int
        |}
}
```

This is an extreme example of course but it would only be a matter of time before someone encounters some problem in the wild.

The type needs to be formatted like this:
```fsharp
            {|
                hash: string
                data: string[]
                otherThing: int
            |}
                ->
                // comment
                (int -> int)
                -> {|
                    hash: string
                    data: string[]
                    otherThing: int
                |}
```
in order to remain valid.